### PR TITLE
fix(FeedSourceSummary): Expose more fields for UI enhancement.

### PR DESCRIPTION
**Pull request by binh-dam-ibigroup**
### Checklist

- [x] Appropriate branch selected _(all PRs must first be merged to `dev` before they can be merged to `master`)_
- [na] Any modified or new methods or classes have helpful JavaDoc and code is thoroughly commented
- [x] The description lists all applicable issues this PR seeks to resolve
- [na] The description lists any configuration setting(s) that differ from the default settings
- [x] All tests and CI builds passing

### Description

Partially addresses https://github.com/ibi-group/datatools-ui/issues/791 by exposing more fields from `FeedVersion` in `FeedVersionSummary`.

### CODE REVIEW

File: src/main/java/com/conveyal/datatools/manager/models/FeedVersion.java [hunk 1]
```diff
@@ -120,7 +120,6 @@ public FeedVersion() {
     /**
      * The feed source this is associated with
      */
-    @JsonView(JsonViews.DataDump.class)
```
Comment (miles-grant-ibigroup) [inline src/main/java/com/conveyal/datatools/manager/models/FeedVersion.java:123]: Why is this removed?

**Review by miles-grant-ibigroup — APPROVED** [review]
Works for me!

File: src/main/java/com/conveyal/datatools/manager/models/FeedVersion.java (hunk 1 above)
Comment (binh-dam-ibigroup) [reply src/main/java/com/conveyal/datatools/manager/models/FeedVersion.java:123]: That field is not included otherwise except in dumps. The UI "Delete Version" button in https://github.com/ibi-group/datatools-ui/pull/792 expects the `feedSourceId` field.

Comment (miles-grant-ibigroup) [reply src/main/java/com/conveyal/datatools/manager/models/FeedVersion.java:123]: Ah this makes sense thank you

File: src/main/java/com/conveyal/datatools/manager/models/FeedVersionSummary.java [hunk 2]
```diff
@@ -11,14 +18,46 @@ public class FeedVersionSummary extends Model implements Serializable {
...
     public int version;
+    public String feedSourceId;
     public String name;
     public String namespace;
     public String originNamespace;
     public Long fileSize;
     public Date updated;
+    /** Only a subset of the validation results are serialized to JSON via getValidationSummary. */
+    @JsonIgnore
+    public ValidationResult validationResult;
```
Comment (binh-dam-ibigroup) [inline src/main/java/com/conveyal/datatools/manager/models/FeedVersionSummary.java:29]: I reversed course for using raw `validationResult`, it has too much stuff imo, so with https://github.com/ibi-group/datatools-server/pull/464/commits/a6a72a7890847b13b4d7b604b09b04427e62c974, I am just exposing two fields from `FeedVersion.validationSummary` that are used in the UI (see https://github.com/ibi-group/datatools-ui/pull/792/commits/915e5afeacebe24b76934b2a3855715baa1bbbfb), any questions let me know @miles-grant-ibigroup.

**Review by philip-cline — APPROVED** [review]
